aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java

```diff
@@ -705,19 +705,15 @@ JSONObject getHostedUIJSON() {
     JSONObject getHostedUIJSON(final AWSConfiguration awsConfig) {
         try {
             JSONObject hostedUIJSONFromJSON = getHostedUIJSONFromJSON(awsConfig);
-            if (hostedUIJSONFromJSON == null) {
-                return null;
-            }
-
             final String hostedUIString = mStore.get(HOSTED_UI_KEY);
             JSONObject hostedUIJSON = null;
             try {
                 hostedUIJSON = new JSONObject(hostedUIString);
             } catch (Exception e) {
                 Log.w(TAG,
-                        "Failed to parse HostedUI settings from store. Defaulting to awsconfiguration.json", e);
+                        "Failed to parse HostedUI settings from store", e);
             }
-            if (hostedUIJSON == null && hostedUIJSONFromJSON != null) {
+            if (hostedUIJSONFromJSON != null && (hostedUIJSON == null || hostedUIJSON.toString() != hostedUIJSONFromJSON.toString())) {
```

Comment:
What happens if the 2 config objects are equal?

Reply:
Yes, the one in the cache is returned in that case.